### PR TITLE
Use `amount.Amount` type in `LiveDB` interface

### PR DESCRIPTION
**Pull request by Mike-CZ**
This PR uses `amount.Amount` type in `LiveDB` interface. Should be merged after [this](https://github.com/Fantom-foundation/Carmen/pull/908).

### CODE REVIEW

File: go/database/mpt/state_test.go [hunk 1]
```diff
@@ -403,7 +404,7 @@ func TestState_StateModificationsWithoutErrorHaveExpectedEffects(t *testing.T) {
 				t.Errorf("account should not exist: %v err: %s", exists, err)
 			}
 
-			var emptyBalance common.Balance
+			emptyBalance := amount.New()
```
Comment (kjezek) [inline go/database/mpt/state_test.go:407]: perhaps we should have also an alias `amont.Zero()`

**Review by kjezek — APPROVED** [review]
looks good to me, thanks

**Review by HerbertJordan — APPROVED** [review]
Looks good, thanks!

File: go/database/mpt/state_test.go (hunk 1 above)
Comment (Mike-CZ) [reply go/database/mpt/state_test.go:407]: Once it's all merged, I'll create a new PR with `Zero` factory method and replaced all `New()` instances.

**Review by kjezek — APPROVED** [review]

**Pull request merged**